data_store/elastic.go

```
@@ -8,87 +8,114 @@ import (
+	if err != nil {
+		return nil, err
+	}
+
+	return &book, nil
 }
 
-func (e *Elastic) InsertBook(book *models.Book) (string, error) {
+func (e *Elastic) InsertBook(book models.Book) (string, error) {
 	ctx := context.Background()
-	putStatus, err := (e.Client).Index().Index(INDEX_NAME).BodyJson(*book).Do(ctx)
+	response, err := (e.Client).Index().Index(INDEX_NAME).BodyJson(book).Do(ctx)
 	if err != nil {
 		return "", err
 	}
-	return putStatus.Id, nil
+
+	return response.Id, nil
 }
+
 func (e *Elastic) UpdateBook(title string, id string) error {
 	ctx := context.Background()
 	_, err := (e.Client).Update().Index(INDEX_NAME).Id(id).Doc(map[string]interface{}{"title": title}).Do(ctx)
 	if err != nil {
 		return err
 	}
+
 	return nil
 }
+
 func (e *Elastic) DeleteBook(id string) error {
 	ctx := context.Background()
-	_, err := (e.Client).Delete().Index("books_yuval").Id(id).Do(ctx)
+	_, err := (e.Client).Delete().Index(INDEX_NAME).Id(id).Do(ctx)
 	if err != nil {
 		return err
 	}
+
 	return nil
 }
 
 func (e *Elastic) SearchBook(title string, authorName string, priceRangeStr string) ([]models.Book, error) {
 	ctx := context.Background()
 	query := elastic.NewBoolQuery()
-	if FieldExist(title) {
-		matchQuery1 := elastic.NewMatchQuery("title", title)
-		query.Filter(matchQuery1)
+	if title != "" {
+		query.Filter(elastic.NewMatchQuery("title", title))
 	}
-	if FieldExist(authorName) {
-		matchQuery2 := elastic.NewMatchQuery("author_name", authorName)
-		query.Filter(matchQuery2)
+	if authorName != "" {
+		query.Filter(elastic.NewMatchQuery("author_name", authorName))
 	}
-	if FieldExist(priceRangeStr) {
-		priceRange := GetPriceRange(priceRangeStr)
-		if priceRange != nil {
-			rangeQuery := elastic.NewRangeQuery("price").From(priceRange[0]).To(priceRange[1])
-			query.Filter(rangeQuery)
+	if priceRangeStr != "" {
+		priceRange, err := GetPriceRange(priceRangeStr)
+		if err != nil {
+			return nil, err
 		}
+		if priceRange == nil {
+			return nil, nil
+		}
```

Are you sure you want this behavior? When will it be nil (if `priceRangeStr` != "") seems like you should return an error